panther_gpiod/test/test_gpiod.cpp

```diff
@@ -0,0 +1,208 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <streambuf>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include "gpiod.hpp"
+
+#include "panther_gpiod/gpio_driver.hpp"
+
+using namespace panther_gpiod;
+
+class TestGPIODriver : public ::testing::Test
+{
+public:
+  TestGPIODriver() = default;
+  virtual ~TestGPIODriver() = default;
+
+  void GPIOEventCallback(const GPIOInfo & gpio_info);
+
+protected:
+  void ReinitializeGPIODriver(std::vector<GPIOInfo> gpio_info_storage);
+  void ReinitializeEventSummary();
+  void SetAndVerifyPinState(const GPIOPin & pin);
+
+  std::unique_ptr<GPIODriver> gpio_driver_;
+  std::pair<GPIOPin, gpiod::line::value> last_gpio_event_summary_;
+  const std::vector<GPIOInfo> gpio_config_info_{
+    GPIOInfo{GPIOPin::LED_SBC_SEL, gpiod::line::direction::OUTPUT},
+  };
+};
+
+void TestGPIODriver::ReinitializeGPIODriver(std::vector<GPIOInfo> gpio_info_storage)
+{
+  gpio_driver_.reset();
+  gpio_driver_ = std::make_unique<GPIODriver>(gpio_info_storage);
+}
+
+void TestGPIODriver::ReinitializeEventSummary()
+{
+  last_gpio_event_summary_.first = static_cast<GPIOPin>(-1);
+  last_gpio_event_summary_.second = static_cast<gpiod::line::value>(-1);
+}
+
+void TestGPIODriver::GPIOEventCallback(const GPIOInfo & gpio_info)
+{
+  last_gpio_event_summary_.first = gpio_info.pin;
+  last_gpio_event_summary_.second = gpio_info.value;
+}
+
+void TestGPIODriver::SetAndVerifyPinState(const GPIOPin & pin)
+{
+  EXPECT_TRUE(gpio_driver_->SetPinValue(pin, true));
+  EXPECT_TRUE(gpio_driver_->IsPinActive(pin));
+
+  EXPECT_TRUE(gpio_driver_->SetPinValue(pin, false));
+  EXPECT_FALSE(gpio_driver_->IsPinActive(pin));
+}
+
+TEST_F(TestGPIODriver, InitGPIODriver_WrongPinConfigFail)
+{
+  // There is no OS version that supports simultaneous operation of MOTOR_ON and VMOT_ON pins.
+  EXPECT_THROW(
+    {
+      try {
+        ReinitializeGPIODriver({GPIOInfo{GPIOPin::MOTOR_ON, gpiod::line::direction::OUTPUT}});
+        ReinitializeGPIODriver({GPIOInfo{GPIOPin::VMOT_ON, gpiod::line::direction::OUTPUT}});
+      } catch (const std::invalid_argument & e) {
+        EXPECT_STREQ("error requesting GPIO lines", e.what());
+        throw;
+      }
+    },
+    std::invalid_argument);
+}
+
+TEST_F(TestGPIODriver, GPIODriverManipulatePin_WrongPinFail)
+{
+  ReinitializeGPIODriver(gpio_config_info_);
+
+  EXPECT_THROW(
+    {
+      try {
+        gpio_driver_->SetPinValue(static_cast<GPIOPin>(-1), true);
```

Comment:
```suggestion
        this->gpio_driver_->SetPinValue(static_cast<GPIOPin>(-1), true);
```
applies to other tests that use TestGPIODriver objects